Gems/Terrain/Code/Source/Components/TerrainPhysicsColliderComponent.cpp

```diff
@@ -414,9 +460,13 @@ namespace Terrain
 
     AZStd::vector<Physics::HeightMaterialPoint> TerrainPhysicsColliderComponent::GetHeightsAndMaterials() const
     {
-        AZStd::vector<Physics::HeightMaterialPoint> heightMaterials;
-        GenerateHeightsAndMaterialsInBounds(heightMaterials);
+        int32_t gridWidth = 0, gridHeight = 0;
+        GetHeightfieldGridSize(gridWidth, gridHeight);
+        AZ_Assert(gridWidth * gridHeight != 0, "GetHeightsAndMaterials: Invalid grid size. Size cannot be zero.");
+
+        AZStd::vector<Physics::HeightMaterialPoint> heightMaterials(gridWidth * gridHeight);
+        UpdateHeightsAndMaterials(heightMaterials);
 
-        return heightMaterials;
+        return AZStd::move(heightMaterials);
```

Comment:
nit: no need to use AzStd::move in the final return statement, it will automatically be moved.